### PR TITLE
[Testing][Refactoring] ModemMonitorServiceImpl tests and refactoring

### PR DESCRIPTION
**Pull request by markoer**
Added service activation tests. Refactored so that testing was enabled.
Code review/actual test would be appreciated due to changed executor initialization.

@ibinshtok please review.

### CODE REVIEW

**Review by ibinshtok — CHANGES_REQUESTED** [review]
@markoer should we invoke the submitMonitorTask() from the processNetworkConfigurationChangeEvent() method as well?

File: kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/monitor/ModemMonitorServiceImpl.java [hunk 1]
```diff
@@ -168,6 +168,9 @@ protected void activate(ComponentContext componentContext) {
 
         stopThread = new AtomicBoolean();
 
+        // must be initialized before trackModem() is called; risk of NPE otherwise
+        this.m_executor = Executors.newSingleThreadExecutor();
```
Comment (MMaiero) [inline kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/monitor/ModemMonitorServiceImpl.java]: If it does not go ever 1k lines, I believe we can also rename the variables according to the new conventions.

Comment (markoer) [reply kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/monitor/ModemMonitorServiceImpl.java]: I need to make updates anyway, so, yes.

**Comment by MMaiero** [comment]
@ibinshtok Ok for you to merge?

**Review by ibinshtok — APPROVED** [review]

**Comment by MMaiero** [comment]
@ibinshtok I believe something got broken with your additions: now we have double commits for the same thing by @markoer. Probably you needed to update your branch with upstream before pushing the new commits. Anyway, it should be working.

**Comment by ibinshtok** [comment]
@MMaiero Yes, it should work. It happened because I forgot to do 'git pull origin' before adding a few more changes and  you have modified the code as well.

**Pull request merged**